src/main/java/com/farao_community/farao/core_cc/adapter/service/CoreCCAdapterService.java

```diff
@@ -39,50 +43,71 @@
+        this.streamBridge = streamBridge;
+    }
+
+    public void handleTask(final TaskDto taskDto, final boolean isLaunchedAutomatically) {
+        handleTask(taskDto, isLaunchedAutomatically, null);
     }
 
-    public void handleTask(TaskDto taskDto, boolean isLaunchedAutomatically) {
+    public void handleTask(final TaskDto taskDto, final boolean isLaunchedAutomatically, final List<TaskParameterDto> parameters) {
         final String runMode = isLaunchedAutomatically ? "automatic" : "manual";
-        final OffsetDateTime taskTimestamp = taskDto.getTimestamp();
+        final String timestamp = taskDto.getTimestamp().toString();
         try {
-            LOGGER.info("Handling {} run request on TS {} ", runMode, taskTimestamp);
-            List<ProcessFileDto> inputFiles = getInputProcessFilesFromRaoRequest(taskDto);
-
-            eventsLogger.info("Task launched on TS {}", taskTimestamp);
-            taskManagerService.updateTaskStatusToPending(taskTimestamp);
-            taskManagerService.addNewRunInTaskHistory(taskTimestamp, inputFiles);
-            TaskDto updatedTaskDto = taskManagerService.getUpdatedTask(taskTimestamp);
-            final CoreCCRequest coreCCRequest = getCoreCCRequest(updatedTaskDto, inputFiles, isLaunchedAutomatically);
-            runAsync(coreCCRequest);
+            LOGGER.info("Handling {} run request on TS {} ", runMode, timestamp);
+            final List<ProcessFileDto> inputFiles = getInputProcessFilesFromRaoRequest(taskDto);
+            final Optional<TaskDto> taskDtoWithRunOpt = taskManagerService.addNewRunInTaskHistory(timestamp, inputFiles);
+            if (taskDtoWithRunOpt.isPresent()) {
+                TaskDto taskDtoWithRun = taskDtoWithRunOpt.get();
+                if (parameters != null && !parameters.isEmpty()) {
+                    taskDtoWithRun = new TaskDto(taskDtoWithRun.getId(), taskDtoWithRun.getTimestamp(), taskDtoWithRun.getStatus(), taskDtoWithRun.getInputs(), taskDtoWithRun.getAvailableInputs(), taskDtoWithRun.getOutputs(), taskDtoWithRun.getProcessEvents(), taskDtoWithRun.getRunHistory(), parameters);
+                }
+
+                final boolean taskStatusUpdated = taskManagerService.updateTaskStatus(timestamp, TaskStatus.PENDING);
+                if (taskStatusUpdated) {
+                    eventsLogger.info("Task launched on TS {}", timestamp);
+                    final CoreCCRequest coreCCRequest = getCoreCCRequest(taskDtoWithRun, inputFiles, isLaunchedAutomatically);
+                    runAsync(coreCCRequest);
+                } else {
+                    eventsLogger.warn("Failed to launch task on TS {}: could not set task's status to PENDING", taskDto.getTimestamp());
+                    streamBridge.send(TASK_STATUS_UPDATE, new TaskStatusUpdate(taskDto.getId(), TaskStatus.ERROR));
+                }
+            } else {
+                eventsLogger.warn("Failed to launch task on TS {}: could not add new run to the task", taskDto.getTimestamp());
```

Comment:
Perhaps use the 'timestamp' variable here and line 88, instead of retrieving the timestamp from taskDto again, as you have done line 84